devel/gitlab-runner/Portfile

```diff
@@ -19,24 +20,37 @@ long_description    GitLab Runner is the open source project that is used to \
 
 homepage            https://docs.gitlab.com/runner/
 
-master_sites        https://gitlab.com/gitlab-org/gitlab-runner/-/archive/v${version}/
-distname            gitlab-runner-v${version}
-
-checksums           rmd160  f362a2eed88aa25bfaf15ffc6d378610c02838e9 \
-                    sha256  2700a09e1bb468cee8abc01998f7cd99c51f9218e3d0f5d3dcbf334322669869 \
-                    size    9144873
+master_sites        https://gitlab.com/gitlab-org/gitlab-runner/-/releases/v${version}/downloads/binaries/
+
+switch ${build_arch} {
+    x86_64 {
+        distname            gitlab-runner-darwin-amd64
+        checksums           rmd160  7bdfd8bde4bec0fe7b9e21c965fdf90bfdee4577 \
+                            sha256  a1315aff96997c1ac9e84ad38617cfba7d1a8f628920f5a94884c168efb7fb6b \
+                            size    49973200
+    }
+    arm64 {
+        distname            gitlab-runner-darwin-arm64
+        checksums           rmd160  5b3121c39636c93f5e395d3f04329b7abf44822c \
+                            sha256  81d1ead766b26b0452195424ef059a2d2780164d095900530de5a05508322f1d \
+                            size    51837138
+    }
+    default {
+        known_fail yes
+        pre-fetch {
+            ui_error "${subport} @ ${version} only supported for architectures ${supported_archs}"
+            return -code error "Unsupported architecture: ${build_arch}"
+        }
+    }
+}
 
-# Reproduce the "build_simple" target from the upstream Makefile
-set go_ldflags      "-X ${go.package}/common.NAME=${go.package} \
-    -X ${go.package}/common.VERSION=${version} \
-    -X ${go.package}/common.REVISION=unknown \
-    -X ${go.package}/common.BUILT=unknown \
-    -X ${go.package}/common.BRANCH=unknown \
-    -s -w"
-build.args          -ldflags \"${go_ldflags}\" -o out/binaries/${name} ${go.package}
+distfiles           ${distname}
+extract {}
+use_configure       no
+build {}
```

Comment:
It can probably be built from source, but I’m not very familiar with Go and I’ve stopped maintaining this port, because I’m no longer interested in using it.

Reply:
Here's an example of this change building from source: https://github.com/macports/macports-ports/pull/15751